src/api/StepsExchanger.ts

```diff
@@ -56,41 +91,106 @@ export class StepsExchanger extends CollabProvider {
     updateStoreVersion: (version: number) => void
   ) {
     super()
+
     this.projectID = projectID
     this.manuscriptID = manuscriptID
     this.currentVersion = currentVersion
     this.isThrottling = new ObservableBoolean()
+    this.saveStatus = new ObservableString()
     this.debounce = saveWithDebounce()
     this.api = api
-    this.start()
     this.updateStoreVersion = updateStoreVersion
+    // Initialize closeConnection as no-op to ensure it's always defined
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    this.closeConnection = () => {}
+
+    if (!navigator.onLine) {
+      this.saveStatus.setValue('offline')
+    } else {
+      this.start()
+    }
+
+    this.addNetworkListeners()
     this.stop = this.stop.bind(this)
   }
 
+  private addNetworkListeners() {
+    window.addEventListener('online', this.handleOnline)
+    window.addEventListener('offline', this.handleOffline)
+  }
+
+  private removeNetworkListeners() {
+    window.removeEventListener('online', this.handleOnline)
+    window.removeEventListener('offline', this.handleOffline)
+  }
+
+  private handleOnline = () => {
+    console.log('Connection recovered. Going back online.')
+    this.saveStatus.setValue('idle')
+    this.start()
+  }
+
+  private handleOffline = () => {
+    console.log('Connection lost. Going offline.')
+    this.saveStatus.setValue('offline')
+    this.closeConnection()
+  }
+
   async sendSteps(
     version: number,
     steps: Step[],
     clientID: string,
     flush = false
   ) {
+
     this.flushImmediately = this.debounce(
       async () => {
-        const response = await this.api.sendSteps(
-          this.projectID,
-          this.manuscriptID,
-          {
-            steps: steps,
-            version,
-            clientID,
+        this.saveStatus.setValue('saving')
+
+        const abortController = new AbortController()
```

Comment:
we should not create an abort on debounce, that should be a reference in StepsExchanger so in the subsequent request if there is a request that needs to be aborted we do it